api/src/test/java/org/openmrs/util/jupiter/H2DatabaseIT.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.util.jupiter;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.List;
+
+import liquibase.Contexts;
+import liquibase.Liquibase;
+import liquibase.database.Database;
+import liquibase.database.DatabaseFactory;
+import liquibase.database.jvm.JdbcConnection;
+import liquibase.exception.LiquibaseException;
+import liquibase.resource.ClassLoaderResourceAccessor;
+import org.h2.jdbc.JdbcSQLNonTransientException;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.openmrs.liquibase.LiquibaseProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class H2DatabaseIT implements LiquibaseProvider {
```

Comment:
@WolfSchlegel 

@achilep is working on migrating all openmrs-core tests to JUnit 5. We will need to support JUnit 4 tests for module developers for a while. That is why we sometimes have to duplicate/migrate test helpers (or base classes) like this one so we have a JUnit 4 version for module devs and a JUnit 5 version that we can directly use in openmrs-core. One day when modules have migrated over to JUnit 5 we can remove the JUnit 4 versions.

I am wondering how this affects the `H2DatabaseIT`. Is that theoretically only used in openmrs-core? In what openmrs release was it added?

I am asking so we can either move it to JUnit 5 if thats only internal.
Or whether we have to provide a JUnit 4 and JUnit 5 version since module developers or others are interested in this class. In that case I would like to know if its ok to move it to package `org.openmrs.test` since this is where we have the test helpers/base classes like `BaseContextSensitiveTest` and so on. For JUnit 5 we created a subpackage `org.openmrs.test.jupiter` where the JUnit 5 versions of lets say `BaseContextSensitiveTest` are. This is where I would see this `H2DatabaseIT` in case that has to be supported in JUnit 4 & 5.

Please let us know what you think :)
Thanks a lot!

Reply:
>> Is that theoretically only used in openmrs-core? In what openmrs release was it added?

The test class was added in the context of https://issues.openmrs.org/browse/TRUNK-4830 and merged to master in mid May 2020.

It is safe to assume that this class is only used internally.

The class can be moved to a different package but the class name should not change as the name is used to distinguish between different kinds of integration test, please search for `*DatabaseIT` in `openmrs-core/pom.xml`

Reply:
thanks @WolfSchlegel